traffic_ops/traffic_ops_golang/api/generic_crud.go

```diff
@@ -80,8 +81,33 @@ func GenericCreate(val GenericCreator) (error, error, int) {
 	} else if rowsAffected > 1 {
 		return nil, errors.New("too many ids returned from " + val.GetType() + " insert"), http.StatusInternalServerError
 	}
+
 	val.SetKeys(map[string]interface{}{"id": id})
 	val.SetLastUpdated(lastUpdated)
+
+	if r, ok := val.(Reader); ok {
+		return ReadBack(r)
+	}
+
+	return nil, nil, http.StatusOK
+}
+
+func ReadBack(val Reader) (error, error, int) {
+
+	i := val.(Identifier)
+	keys, _ := i.GetKeys()
+	inf := val.APIInfo()
+	inf.Params["id"] = strconv.Itoa(keys["id"].(int))
+
+	results, usrErr, sysErr, errCode := val.Read()
+	if usrErr != nil || sysErr != nil {
+		return usrErr, sysErr, errCode
+	}
+	if len(results) != 1 {
+		return nil, errors.New("bad result in reading back a POST"), http.StatusInternalServerError
+	}
+	ConvertFrom(&val, results[0])
```

Comment:
This seems like a lot of work to get at the underlying data - maybe instead find a way to expose that?

Reply:
What do you mean `expose`? We are given an interface so we don't have many options.

Reply:
You can always just make them [define a setter for the underlying data](https://play.golang.org/p/02kL8msfpNj)